security/security.service.ts

```diff
@@ -0,0 +1,199 @@
+      .map((token) => {
+        return { token };
+      });
+  }
+
+  public getTokensFromArray(array: CustomActionSchema[] | WebhookResponseSchema[]): string[] {
+    const tokens: string[] = [];
+    for (const { headers } of array) {
+      if (headers && headers['auth-token']) {
+        tokens.push(headers['auth-token']);
+      }
+    }
+    return tokens;
+  }
+
+  public async refreshTokensInWHandCA(
+    iconikService: IconikService,
+    webHooks: WebhookResponseSchema[],
+    customActions: CustomActionSchema[],
+    token: string
+  ): Promise<void> {
+    const refreshTokensInCAs = customActions.map(async (CA) => await this.refreshTokenInCA(token, CA, iconikService));
+    const refreshTokensInWHs = webHooks.map(async (WH) => await this.refreshTokenInWH(token, WH, iconikService));
+
+    await Promise.all([refreshTokensInCAs, refreshTokensInWHs]);
+  }
+
+  private async refreshTokenInCA(token: string, CA: CustomActionSchema, iconikService: IconikService) {
+    try {
+      const { context, id, title, url } = CA;
+      await iconikService.assets.updateCustomAction(context, id!, {
+        context,
+        title,
+        url,
+        headers: { ...CA?.headers, 'auth-token': token },
+      });
+    } catch (error) {
+      console.log('updateCustomAction: ', error);
+    }
+  }
+
+  private async refreshTokenInWH(token: string, WH: WebhookResponseSchema, iconikService: IconikService) {
+    const { event_type, url, id, status, headers } = WH;
+    try {
+      await iconikService.notifications.updateWebhook(id, {
+        url,
+        event_type,
+        status,
+        headers: { ...headers, 'auth-token': token },
+      });
+    } catch (error) {
+      console.log('updateWebhook: ', error);
+    }
+  }
+
+  public async createRuleAndBindLambda(
+    ruleName: string,
+    ruleSchedule: number,
+    lambdaARN: string
+  ): Promise<PutRuleResponse | void> {
+    const params: PutRuleRequest = {
+      Name: ruleName,
+      ScheduleExpression: `rate(${ruleSchedule} minutes)`,
+      State: 'ENABLED',
+    };
+
+    const isRule = await this.getCloudWatchRule(ruleName);
+    if (isRule) {
+      return await this.putCloudWatchRule(params);
+    }
+
+    const { RuleArn } = await this.putCloudWatchRule(params);
+
+    await this.addFunctionInvokePermissionForRule(`${ruleName}-permission`, lambdaARN, RuleArn!);
+
+    await this.putCloudWatchTargets({
+      Rule: ruleName,
+      Targets: [
+        {
+          Id: '1',
+          Arn: lambdaARN,
+        },
+      ],
+    });
+  }
+
+  public async getTokensFromDynamoDB(): Promise<ScanResponse<IconikToken>> {
+    return (await IconikTokenModel.scan().exec()) as ScanResponse<IconikToken>;
+  }
+
+  public isInvalidateIconikToken(dateOfCreation: Date) {
+    const hourInMilliseconds = 3600000;
+    const refreshTokenMilliseconds: number = parseFloat(getEnv('INVALIDATE_TOKEN_DELAY_HOURS')) * hourInMilliseconds;

```

Comment:
Should you check the value for NaN and `value > 0`before use it?

Reply:
the default value is already in env (1 hour) also in README.md there is a description of how to use it, so this value should not be 0

Reply:
So, we have default value for ENV var and some notes in readme. Do you think this prevent from a mistake?